package/kedro_viz/server.py

```diff
@@ -471,10 +472,74 @@ def format_pipeline_data(
             nodes[node_id]["pipelines"].append(pipeline_key)
 
 
+def _get_dataset_node(node_id, namespace):
+    if KEDRO_VERSION.match(">=0.16.0"):
+        try:
+            dataset = _CATALOG._get_dataset(namespace)
+        except DataSetNotFoundError:
+            dataset = None
+    else:
+        dataset = _CATALOG._data_sets.get(namespace)
+    _NODES[node_id] = {"type": "data", "obj": dataset}
+
+
 @app.route("/api/nodes.json")
 def nodes_json():
     """Serve the pipeline data."""
-    return jsonify(data)
+    return jsonify(_DATA)
+
+
+@app.route("/api/nodes/<string:node_id>")
+def nodes_metadata(node_id):
+    """Serve the metadata for node and dataset."""
+    node = _NODES.get(node_id)
+    if not node:
+        abort(404, description="Invalid node ID.")
+    if node["type"] == "task":
+        task_metadata = _get_task_metadata(node)
+        return jsonify(task_metadata)
+    if node["type"] == "data":
+        dataset_metadata = _get_dataset_metadata(node)
+        return jsonify(dataset_metadata)
+
+    # return empty JSON for parameters type
+    return jsonify({})
+
+
+def _get_task_metadata(node):
+    """Get a dictionary of task metadata: 'code', 'code_location' and 'docstring'.
+    For 'code_location', remove the path to the project from the full code location
+    before sending to JSON.
+
+    Example:
+        'code_full_path':   'path-to-project/project_root/path-to-code/node.py'
+        'Path.cwd().parent':'path-to-project/'
+        'code_location':    'project_root/path-to-code/node.py''
+
+    """
+    task_metadata = {"code": inspect.getsource(node["obj"]._func)}
+
+    code_full_path = Path(inspect.getfile(node["obj"]._func)).expanduser().resolve()
+    code_location = code_full_path.relative_to(Path.cwd().parent)
```

Comment:
I still don't get why it's a parent of the working dir and not the working dir itself 😄

Reply:
Let me rephrase this, as the line itself doesn't explain much. 

Does this example help?

![Screenshot 2020-08-07 at 11 07 27](https://user-images.githubusercontent.com/8097799/89635074-51618780-d89e-11ea-8a8a-03e5e5234130.png)

and this. Without `parent`, it starts from `src`, but we want do display from `new_api/src/`
![Screenshot 2020-08-07 at 11 08 13](https://user-images.githubusercontent.com/8097799/89635232-938ac900-d89e-11ea-85ca-845822cb9a4e.png)

Reply:
Why do we need to show `new_api`? As to me, `src/pipelines/...` is exactly what we need.

Reply:
Okay, I can do that :) If someone (or Yetu) asks us to show from the project root, we can easily add `parent` later.